### ISSUE TITLE
classes with member functions triggers U2U1001

### ISSUE THREAD

**Issue by jnyrup** [u2uconsult/codeanalyzers#2]
While `MyClass` _can_ be made static, applying the `static` keyword to `MyClass` before `MyMethod` i made static produces invalid code.

```
class MyClass // triggers U2U1001
    {
        public void MyMethod() { }
    }
```

**Comment by KrisVandermotten** [comment]
Indeed, making such a class static requires all members to be made static as well.

What exactly is the issue you're trying to raise?

**Comment by jnyrup** [comment]
I'm sorry for not being clear enough.
The analysis is of course correct that `MyMethod` can be static and _then_ `MyClass` can be static.
The codefix on `MyClass` makes _only_ the class static, which leaves it in an invalid state as `MyMethod` is still an instance member.

So the issue I'm raising is, that if you strive to leave the code in valid states after applying any codefix, here's a case where it doesn't.

**Comment by KrisVandermotten** [comment]
We are testing a code fix that will do many necessary transformations automatically.

For example,

    class C
    {
        protected int M() => 0;

        public int M(int i) => this.M() + i;
    }

is transformed into

    static class C
    {
        private static int M() => 0;

        public static int M(int i) => M() + i;
    }

**Comment by jnyrup** [comment]
That sounds awesome!

**Comment by KrisVandermotten** [comment]
The code fix has been included in version 2.6.
